vpd-tool/src/vpd_tool_main.cpp

```diff
@@ -67,19 +88,48 @@ int main(int argc, char** argv)
         return l_rc;
     }
 
-    if (*l_hardwareFlag && !std::filesystem::exists(l_vpdPath))
-    {
-        std::cerr << "Given EEPROM file path doesn't exist : " + l_vpdPath
-                  << std::endl;
-        return l_rc;
-    }
-
-    (void)l_fileOption;
-
     if (*l_readFlag)
     {
         // TODO: call read keyword implementation from here.
     }
+    else if (l_writeFlag->count() > 0)
+    {
+        if ((l_hardwareFlag->count() > 0) &&
+            !std::filesystem::exists(l_vpdPath))
+        {
+            std::cerr << "Given EEPROM file path doesn't exist : " + l_vpdPath
+                      << std::endl;
+            return l_rc;
+        }
+
+        if ((l_fileOption->count() > 0) && !std::filesystem::exists(l_filePath))
+        {
+            std::cerr
+                << "Please provide a valid absolute file path which has keyword value.\nUse --value/--file to give "
+                   "keyword value. Refer --help."
+                << std::endl;
+            return l_rc;
+        }
+        else if ((l_keywordValueOption->count() > 0) && l_keywordValue.empty())
+        {
+            std::cerr
+                << "Please provide keyword value.\nUse --value/--file to give "
+                   "keyword value. Refer --help."
+                << std::endl;
+            return l_rc;
+        }
+        else if ((l_fileOption->count() == 0) &&
```

Comment:
you should have this validation immediately after line 95.

Reply:
file option currently used under read & write operation,
we should restrict its validation and usage under the read/write operations only.